### PR TITLE
Remove KOKKOS_THREAD_LOCAL

### PR DESCRIPTION
**Pull request by masterleinad**
There doesn't really seem any need to define this macro when we can use `thread_local` instead.

### CODE REVIEW

**Review by PhilMiller — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

**Comment by crtrott** [comment]
How sure are we that no one downstream uses that? Maybe deprecate for 3.7 and then remove (could define it to KOKKOS_THREAD_LOCAL = thread_local?

**Review by crtrott — DISMISSED** [review]
Should we deprecate? Its a kinda public name?

**Comment by dalg24** [comment]
OK with what Christian suggest.
Keep when deprecated code is enabled.  I suspect you might be able to slap the deprecated attribute in the front when warning are enabled.

**Comment by masterleinad** [comment]
> How sure are we that no one downstream uses that? Maybe deprecate for 3.7 and then remove (could define it to KOKKOS_THREAD_LOCAL = thread_local?

It's not used in `Trilonos` at least but I kept it now with deprecated code on.

File: core/src/Kokkos_Macros.hpp [hunk 1]
```diff
@@ -652,6 +652,17 @@ static constexpr bool kokkos_omp_on_host() { return false; }
...
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_3
+#if (defined(KOKKOS_COMPILER_MSVC) && !defined(KOKKOS_COMPILER_CLANG)) || \
+    (defined(KOKKOS_COMPILER_INTEL) && defined(_WIN32))
+#define KOKKOS_THREAD_LOCAL \
+  KOKKOS_DEPRECATED_WITH_COMMENT("Use thread_local instead!") __declspec(thread)
+#else
+#define KOKKOS_THREAD_LOCAL \
+  KOKKOS_DEPRECATED_WITH_COMMENT("Use thread_local instead!") __thread
+#endif
```
Comment (dalg24) [inline core/src/Kokkos_Macros.hpp:663]:
```suggestion
  KOKKOS_DEPRECATED_WITH_COMMENT("Use thread_local instead!") thread_local
```

**Comment by dalg24** [comment]
Not sure what is going on with Jenkins but that commit was built successfully so I am going to merge as is
https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/8864/pipeline

**Pull request merged**